### PR TITLE
Namespaced + Enhanced Maths System

### PR DESCRIPTION
**Pull request by beeperdeeper089**
**Addresses: #15**

**Authors: @beeperdeeper089**

## Summary of changes
  - Copy a templated maths system into Math/
  - Make access easier through the usage of ``math::vec2/vec3/mat4`` with functionality such as:
    - ``math::vec2 (for float)``
    - ``math::vec2i (for integer)``
    - ``math::vec2f (for float)``
    - ``math::vec2d (for double)``
    - ``math::vec2u (for unsigned int)``
    - This scheme applies to the ``vec3`` as well.
  
## Caveats
  - Vector2/3 are templated, leading to potentially longer compile times, but this should be negligible.

## On approval
Merge.

### CODE REVIEW

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- Sandbox/Include/Sandbox/Sandbox.hpp  1
- Engine/Include/Quartz2/Math/Vector3.hpp  7
- Engine/Include/Quartz2/Math/MathUtils.hpp  1
- Engine/Include/Quartz2/Math/Matrix4x4.hpp  1
- Sandbox/Source/Main.cpp  1
- Engine/Source/Math/Matrix4x4.cpp  5
- Engine/Include/Quartz2/Math/Vector2.hpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803327&bid=15983186)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- Engine/Include/Quartz2/Math/Vector2.hpp  2
- Engine/Source/Math/Matrix4x4.cpp  5
- Engine/Include/Quartz2/Math/Matrix4x4.hpp  1
- Engine/Include/Quartz2/Math/Vector3.hpp  7
- Engine/Include/Quartz2/Math/MathUtils.hpp  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803327&bid=15983186)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- Engine/Source/Math/Matrix4x4.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803327&bid=15983186)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- Engine/Source/Math/Matrix4x4.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803327&bid=15983186)

**Comment by apachano** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Clones added
============
- Engine/Source/Math/Matrix4x4.cpp  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GentenStudios/Phoenix/pullRequest?prid=4803327&bid=15983186)

**Comment by apachano** [comment]
WIP?

**Comment by beeperdeeper089** [comment]
Yup, WIP, I'll update the title.

**Comment by apachano** [comment]
@beeperdeeper089 Merge develop in so files are easy to review, I haven't looked at code yet but since we use math everywhere, please add super basic doxy comments just to files, classes, variables, and functions. Classes need a short description of what they are (a brief or slightly more than a brief) and functions need a brief, arguments, and return value documented. Variables should be a single-line brief

File: Engine/Include/Quartz2/Math/Math.hpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019-20 Genten Studios
```
Comment (M3TIOR) [inline Engine/Include/Quartz2/Math/Math.hpp]: Whoops, looks like you invalidated some of Toby's copyright stuff.

File: Engine/Include/Quartz2/Math/MathUtils.hpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019-20 Genten Studios
```
Comment (M3TIOR) [inline Engine/Include/Quartz2/Math/MathUtils.hpp]: Ditto previous

File: Engine/Include/Quartz2/Math/Matrix4x4.hpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2019-20 Genten Studios
```
Comment (M3TIOR) [inline Engine/Include/Quartz2/Math/Matrix4x4.hpp]: Did you forget to pull / merge from develop?

File: Engine/Include/Quartz2/Math/Math.hpp (hunk 1 above)
Comment (beeperdeeper089) [reply Engine/Include/Quartz2/Math/Math.hpp]: Yeah, I'll let all this get merged in I would say and then I'll use [my license automation tool](https://github.com/beeperdeeper089/license-automator) to just set everything to 2019-20

File: Engine/Include/Quartz2/Math/Matrix4x4.hpp (hunk 3 above)
Comment (beeperdeeper089) [reply Engine/Include/Quartz2/Math/Matrix4x4.hpp]: Did that but im a doofus :(

**Comment by M3TIOR** [comment]
LGTM

**Pull request merged**